### PR TITLE
Readme, Example, and typescript binding cleanup

### PR DESCRIPTION
**Pull request by aricart**
- fixed examples usage of `--creds` parameter
- added support for `--queue` group to node-sub
- added tool `node-rep` for botting `node-req` uses
- made the `node-[pub|sub|req|rep]` example tools binaries, this allows developers to use these tools while developing their apps with `npx node-[pub|sub|req|rep]` without additional copying or editing

- unified source code version with its package.json version - this allows the npm version management to work without breaking the tests or requiring manual edit

- simplified typescript definitions for `connect`, definition was missing hybrid `connect(url, opts)`.
- added typescript type `RequestOptions` for request options, these were incorrectly referencing `SubscribeOptions`

- reworked README.md examples formatting, removed example of `requestOne` as it provides no benefit of over `request` when additional arguments are specified. A future version should deprecate this API.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -227,166 +215,171 @@ const sid1 = nc.subscribe(subj, {queue: 'q1'}, () => {
...
 
 ```javascript
-var nats = require('nats');
-var fs = require('fs');
+const NATS = require('nats')
+const fs = require('fs')
 
 // Simple TLS connect
-var nc = nats.connect({port: TLSPORT, tls: true});
+let nc = NATS.connect({ tls: true })
```
Comment (derekcollison) [inline README.md]: Do we need this? If server requires TLS it should just do the right thing yes?

File: examples/node-pub [hunk 2]
```diff
@@ -18,7 +18,8 @@
 /* jslint node: true */
 'use strict'
 
-const nats = require('nats')
+// change '../' to 'nats' when copying
```
Comment (derekcollison) [inline examples/node-pub]: Not sure what this means.

File: examples/node-pub [hunk 3]
```diff
@@ -27,19 +28,19 @@ const subject = argv._[0]
 const msg = argv._[1] || ''
 
 if (!subject) {
-  console.log('Usage: node-pub  [-s server] [--creds file] <subject> [msg]')
+  console.log('Usage: node-pub  [-s server] [--creds=filepath] <subject> [msg]')
```
Comment (derekcollison) [inline examples/node-pub]: X=Y syntax required?

File: examples/node-rep [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+const argv = require('minimist')(process.argv.slice(2))
+
+const url = argv.s || nats.DEFAULT_URI
+const creds = argv.creds
+const queue = argv.queue
+const subject = argv._[0]
+const response = argv._[1] || ''
+
+if (!subject) {
+  console.log('Usage: node-rep  [-s server] [--creds=creds_file_path] [--queue=qname] <subject> [response]')
```
Comment (derekcollison) [inline examples/node-rep]: Made mw think of repl, had to read code. Would make node-reply or nats-reply.

File: README.md (hunk 1 above)
Comment (aricart) [reply README.md]: this is saying that the client requires the connection to be tls.

File: examples/node-pub (hunk 2 above)
Comment (aricart) [reply examples/node-pub]: I made the examples work right from the package when installed as `npx

File: examples/node-pub (hunk 3 above)
Comment (aricart) [reply examples/node-pub]: yes for flags that are not a single letter

File: examples/node-rep (hunk 4 above)
Comment (aricart) [reply examples/node-rep]: Let me rename it

Comment (derekcollison) [reply examples/node-rep]: ok

**Pull request merged**